selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestUserServiceClient.java

```diff
@@ -10,65 +10,69 @@
  */
 package org.eclipse.che.selenium.core.client;
 
-import static org.eclipse.che.dto.server.DtoFactory.newDto;
-
-import com.google.inject.Inject;
-import com.google.inject.Singleton;
-import java.net.URLEncoder;
+import org.eclipse.che.api.core.BadRequestException;
+import org.eclipse.che.api.core.ConflictException;
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.api.core.ServerException;
 import org.eclipse.che.api.core.model.user.User;
-import org.eclipse.che.api.core.rest.HttpJsonRequestFactory;
-import org.eclipse.che.api.core.rest.HttpJsonResponse;
-import org.eclipse.che.api.user.shared.dto.UserDto;
-import org.eclipse.che.selenium.core.provider.TestApiEndpointUrlProvider;
-import org.eclipse.che.selenium.core.requestfactory.TestAdminHttpJsonRequestFactory;
-
-/** @author Musienko Maxim */
-@Singleton
-public class TestUserServiceClient {
-  private final String apiEndpoint;
-  private final HttpJsonRequestFactory requestFactory;
 
-  @Inject
-  public TestUserServiceClient(
-      TestApiEndpointUrlProvider apiEndpointProvider,
-      TestAdminHttpJsonRequestFactory requestFactory) {
-    this.apiEndpoint = apiEndpointProvider.get().toString();
-    this.requestFactory = requestFactory;
-  }
+/**
+ * @author Mihail Kuznyetsov
+ * @author Anton Korneta
+ */
+public interface TestUserServiceClient {
 
-  public User getByEmail(String email) throws Exception {
```

Comment:
We need this method for https://github.com/codenvy/codenvy/blob/master/selenium/codenvy-selenium-test/src/test/java/com/codenvy/selenium/factory/AuthenticateAndAcceptFactoryThroughGitHubOAuthTest.java

Reply:
I'll adapt this code in codenvy PR